### DISCUSSION TITLE
Pull down tabs bar please, to make title bar clean to allow DnD

### DISCUSSION THREAD

**Discussion by alexeik** [sumatrapdfreader/sumatrapdf#2662] [Ideas]
Pull down tabs bar please, to make title bar clean to allow DnD 

u know that,
no title bar , not DnD...
hard to move app windows without clean title bar over 5k monitor.

**Comment by GitHubRulesOK** [comment]
Unclear what your asking for Drag n Drop
You use the lower canvas not the title bar.

![image](https://user-images.githubusercontent.com/12754913/171934365-594f2e41-3b0f-4558-8b9f-52364229135f.png)

To drag the window you use the icon 

![image](https://user-images.githubusercontent.com/12754913/171935187-ed094048-657b-4c0b-8c1d-ec9ea45fb55e.png)

**Comment by alexeik** [comment]
i have 3.2 verision
<img width="1438" alt="image" src="https://user-images.githubusercontent.com/318732/172020637-92ed25b6-e177-4ca5-a23d-897835085a98.png">

**Comment by GitHubRulesOK** [comment]
It was the same in 3.2 but changes can only be made to current pre-release (3.6+)

**Comment by alexeik** [comment]
how i can fix it in 3.2 ???

**Reply by GitHubRulesOK** [reply]
you need to use the drag zones shown here in red (its the top bar or yellow icon for drag SumatraPDF)
![image](https://user-images.githubusercontent.com/12754913/172057811-d10de32f-b2f2-44dc-a9aa-ad0681869d66.png)

**Comment by alexeik** [comment]
i know) but it IS very useless

i want your layout, not ugly like mine.

**Comment by GitHubRulesOK** [comment]
My layout is not supposed to happen but always worked like that, it is consider a temporary state between no-tabs and tabs, it happens when you start without tabs then switch them tabs on, so you get both, just don't complaint it does not work or it may be removed :-)
![image](https://user-images.githubusercontent.com/12754913/172071582-d54e43df-6512-4254-85a9-885c4c198148.png)

**Reply by alexeik** [reply]
so how to pull down ?

**Reply by GitHubRulesOK** [reply]
SumatraPDF tabs do not detach (pull away from the tab bar) for the present they can only slide right or left
Work is in progress to allow tab bar changes but I do not think bar height is likely to change except by any windows setting.

**Reply by alexeik** [reply]
not pull away.)
just pull 1 row down as your screenshots has.
my tabs sitting on title bar - bad.

**Reply by GitHubRulesOK** [reply]
the narrow bar you get is because it is a fixed value of about one dozen pixels so at high dpi it is very thin. That is why the draggable icon was added on the far Left (same as browsers like Firefox etc.) so you can drag the icon. 
![image](https://user-images.githubusercontent.com/12754913/172200197-47afc2de-b77e-4d9b-9dcb-39af20aca847.png)
![image](https://user-images.githubusercontent.com/12754913/172200923-aada8b0f-8fce-4288-8894-4a01383eed67.png)

My view is an incorrect view but can help for getting the bar to appear larger as 2 rows by including the menu bar when first started without tabs on.

**Reply by alexeik** [reply]
ur right i have 4k, 155 ppi
icon not a solution.
always has a need to dnd by title bar on right side, not  by left where icon placed.
what is your origin language?

**Reply by GitHubRulesOK** [reply]
Language should not stop icon from working but it will be much smaller 
so aim for the black centre ***`PDF`***
 you can also use Move but can be more awkward
![image](https://user-images.githubusercontent.com/12754913/172210186-a4967bf9-7091-4111-b15e-8c19dace21a9.png)

**Reply by alexeik** [reply]
i wanna say i almost never has access to ICON :) cause some windows already overlaps it.
i always has access to part of title bar on right side of window

that is why your steps are useless for me

**Reply by tactilis** [reply]
@alexeik
> i wanna say i almost never has access to ICON :) cause some windows already overlaps it. i always has access to part of title bar on right side of window

I don't understand the problem. As soon as you click _anywhere_ in the SumatraPDF window, it will become the top window and the other window(s) will no longer overlap it. You can then move the window using the icon.

I don't see why @GitHubRulesOK's steps are useless

**Reply by alexeik** [reply]
i just need full height title bar.
to DnD in one second without snipering mouse 

like on screenshots from @GitHubRulesOK'

**Reply by GitHubRulesOK** [reply]
@kjk probably less useful to 3.5 but is there a possibility in dpi / settings to add a means to broaden the top most band ?

**Comment by tactilis** [comment]
> that is why your steps are useless for me

> i just need full height title bar.
>to DnD in one second without snipering mouse
>like on screenshots from @GitHubRulesOK'

OMG @alexeik, @GitHubRulesOK has already told you how to do it.

I've just followed the steps. Clearly they are not useless, see: Tabs, Menu bar, Title bar! 

![image](https://user-images.githubusercontent.com/4060092/172236636-03e16fce-c6f0-4f6f-9417-f7de223d33d9.png)

**Reply by alexeik** [reply]
@tactilis  how?
where ?
sorry if i havent seen

**Reply by alexeik** [reply]
OMG
@GitHubRulesOK  got u
done this hack

omg! ppl
sorry if any
but it was hard to get that from sentences

**Comment by alexeik** [comment]
but its really useless
cause if i close and open sumatra it drops normal layout and do layout with thin title bar + Tabs.

**Reply by GitHubRulesOK** [reply]
I said it is not the way it is supposed to work
and only works if you start with no tabs first, 
so you need to switch tabs off when you close 
and that is easy to forget when closing session. 
and will cause as many windows to open next time as there were tabs when closing
I said it is not the way it is supposed to work

**Reply by alexeik** [reply]
ok) got u

**Comment by tactilis** [comment]
Just close the tabs. Don't close the application. It hardly takes up any resources.

**Reply by alexeik** [reply]
i need tabs to read few books simultaneously

**Reply by GitHubRulesOK** [reply]
There is no way to change the code in older versions unless you do it yourself by mod the variables to your liking, The current pre-release has many many changes but is in this case no different in behavior.
The means to drag such a window is as simple as select the current window and use top left icon OR **hotkey** (currently pre-release has an issue in my sandbox ! dang it !) **<kbd>alt</kbd>+<kbd>space</kbd>+<kbd>M</kbd>** OR icon context menu to Move the window.

Try using alt space M to see if that allows drag better but the only fast way with mouse is drag the icon.
If scale is too small consider change the window scale by use of compatibility.

**Comment by bitkleberAST** [comment]
I also do vote for moving the tabs down.
It is quite annoying to aim on a free spot in the title bar to be able to move the window. The suggestion to use the window menu results in the same effort to aim, and requires an additional click if the window is overlapped by other windows.
Using the keyboard is also not an option, as it requires to move your hands to the keyboard when one is already using the mouse.

![grafik](https://user-images.githubusercontent.com/15101747/229344562-ec5f0510-d822-49c0-9f77-55e27e168a63.png)

The best way to move windows is to drag them by their title bar. The problems described in this post are the reason why Microsoft (and other manufacturers) implemented this function like it is now.

How hard is it to change this? Are there any problems caused by UI elements or third party tools that prevent this change?

**Reply by GitHubRulesOK** [reply]
Every app is different but SumatraPDF is fairly mid range not as grabbing as Firefox shown here (which has no small icon) nor as mean as MS Edge some use other approachs such as place tabs over icons or icons over tabs but the aim is for the target to be as small as possible

![image](https://user-images.githubusercontent.com/12754913/229353217-01d28c3f-0655-4558-9074-e1248f8f1e30.png)

**Reply by bitkleberAST** [reply]
Ok, you have a point here :)
I am using Firefox, but was annoyed by their decision to move the tabs into the title bar so I activated the menu bar permanently. 
![grafik](https://user-images.githubusercontent.com/15101747/229369046-fdea0b19-53a4-4155-8f4e-83087ebb75bb.png)
This creates a nice draggable area. :)

The area above the tabs in SumatraPDF, on the other hand, is way to small on Windows as it is probably only around 2-4 pixels wide before the cursor changes to the resize window North-South cursor. If there are only few tabs, there is much space left in the title bar so you can drag the window conveniently. 
![grafik](https://user-images.githubusercontent.com/15101747/229369183-66188c13-7706-4258-b148-79354023001a.png)

But as soon as you open many files, this space disappears and one is left without a chance to move the window using the mouse.

**Reply by GitHubRulesOK** [reply]
the height is minimum 14 pixels except on the far right about twice as many as Edge and all 14 are active (the cursor change halfway up is a windows thang)
![image](https://user-images.githubusercontent.com/12754913/229370301-4cc97d8d-ad92-4b8d-aadb-9cff418cb385.png)
![image](https://user-images.githubusercontent.com/12754913/229370750-80f7875f-654b-4225-a3bd-e01dee6703c5.png)

**Reply by alexeik** [reply]
on 4k 32'' very hard to pick in 14 pixels...

**Reply by GitHubRulesOK** [reply]
thats why the icon is added see https://github.com/sumatrapdfreader/sumatrapdf/issues/423

**Comment by alexeik** [comment]
Edge have Vertical tabs :) and that is why its free to drag.
But it is very complex to move SumatraPDF window.

**Comment by mika76** [comment]
Is there a way to switch off this tabs-in-title-bar feature off permanently? I turn off the "use tabs" in settings but it just comes back on - also it doesn't seem to change anything. I would prefer the normal window with tabs below/or no tabs would be ok too.

**Comment by alexeik** [comment]
@mika76 they dont want to change tabs-in-title-bar...

**Comment by GitHubRulesOK** [comment]
@mika76 
When you "switch off" tabs the state does not change in the current interface without a restart (hinted at above that temporarily you can have both untill restart)

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/86443d42-e998-4967-8c4b-d040f913f0c5)

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/b5f4154b-acc5-41d6-8ff2-189f58ef8d40)

to drag and drop another file use the main window not the now less useful system bar
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/0b378da0-ee51-4bf0-961c-e999201e4dcc)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/f895ff19-d9ac-4d02-8e51-87570f237b4f)

**Comment by mika76** [comment]
Ok I figured out my problem and though I'd share it here - I had used the portable exe and put it in a place where it didn't have access to write its settings file. That's why whenever I changed the settings it didn't do anything. I would recommend that Sumatra writes the settings file to ProgramData or Application Settings instead of just in its current folder.

**Comment by GitHubRulesOK** [comment]
@mika76 
seems odd you place a portable copy in a read only folder that is what the program folders installed version is for.
For portable use from a CD or other read only USB/folder there is an `-appdata foldername` switch 
so simply start with `sumatrapdf -appdata %temp% filename` or `sumatrapdf -appdata %appdata% filename` 
and the settings will be read from temp or appdata directory accordingly.

**Reply by mika76** [reply]
Yeah for sure it was my mistake - but I didn't know it would try store a settings file. Thanks for the info.
